### PR TITLE
Add live theme switching support

### PR DESCRIPTION
**Pull request by timja**
See https://github.com/jenkinsci/theme-manager-plugin/pull/75

All themes are now loaded on every page, the active theme is controlled by a `data` attribute on the `html` element.

fyi @timbrown5 

Pretty much ready now just need to drop core requirement down

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,20 +1,23 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>4.2</version>
+    <version>4.35</version>
```
Comment (timja) [inline pom.xml]:
```suggestion
    <version>4.37</version>
```

File: pom.xml [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
+  <version>${revision}${changelist}</version>
   <packaging>hpi</packaging>
   <properties>
+    <revision>0.4.2</revision>
+    <changelist>-SNAPSHOT</changelist>
+    <gitHubRepo>jenkinsci/material-theme-plugin</gitHubRepo>
     <revision>0.4.1</revision>
     <changelist>-SNAPSHOT</changelist>
-    <jenkins.version>2.289.1</jenkins.version>
+    <jenkins.version>2.336-rc32081.db_eed62c0c0e</jenkins.version>
```
Comment (timja) [inline pom.xml]:
```suggestion
    <jenkins.version>2.336</jenkins.version>
```

File: pom.xml [hunk 3]
```diff
@@ -37,7 +40,7 @@
     <dependency>
       <groupId>io.jenkins.plugins</groupId>
       <artifactId>theme-manager</artifactId>
-      <version>0.1</version>
+      <version>0.7-rc94.26a_5398f092d</version>
```
Comment (timja) [inline pom.xml]:
```suggestion
      <version>1.0</version>
```

File: pom.xml [hunk 4]
```diff
@@ -67,9 +70,9 @@
   </developers>
 
   <scm>
-    <connection>scm:git:git://github.com/jenkinsci/${project.artifactId}-plugin.git</connection>
-    <developerConnection>scm:git:git@github.com:jenkinsci/${project.artifactId}-plugin.git</developerConnection>
-    <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
+    <connection>scm:git:git://github.com/${gitHubRepo}.git</connection>
```
Comment (timja) [inline pom.xml]:
```suggestion
    <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
```

**Comment by timja** [comment]
@timbrown5 FYI material theme is looking a little broken on recent weeklies due to some icon changes :).

But this is unrelated to that

File: pom.xml [hunk 5]
```diff
@@ -1,20 +1,23 @@
...
+  <version>${revision}${changelist}</version>
   <packaging>hpi</packaging>
   <properties>
+    <revision>0.4.2</revision>
+    <changelist>-SNAPSHOT</changelist>
+    <gitHubRepo>jenkinsci/material-theme-plugin</gitHubRepo>
     <revision>0.4.1</revision>
     <changelist>-SNAPSHOT</changelist>
-    <jenkins.version>2.289.1</jenkins.version>
+    <jenkins.version>2.336</jenkins.version>
```
Comment (timbrown5) [inline pom.xml:20]: As this plugin replaces most of the older Jenkins icons This needs to be pinned for a family of releases.
0.3.x - Supports pre 2.289.1 (specifically before `Add modern icons: build status and weather. ([pull 5065](https://github.com/jenkinsci/jenkins/pull/5065), [pull 5392](https://github.com/jenkinsci/jenkins/pull/5392)`)) 
0.4.x - supports `2.289.1`+ (unless there are significant Jenkins Icon changes).
0.5.x - will support the next big UI change (possible https://github.com/jenkinsci/jenkins/pull/6307).

My hope is that by doing this I can fix critical issues for users stuck on older Jenkins versions, whilst also not breaking any new icon changes (although it sounds like it's a bit broken on recent Jenkins releases?).
As such, can you leave this as `2.289.1`, so it's the minimum supported version?

That said, if this version update is required to support dependencies (e.g. Theme manager) then can you change the version to 0.5.0 (and then 0.6.x will need to support the next UI change).

**Review by timbrown5 — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 6]
```diff
@@ -1,20 +1,23 @@
...
+    <version>4.37</version>
     <relativePath />
   </parent>
   <groupId>io.jenkins.plugins</groupId>
   <artifactId>material-theme</artifactId>
-  <version>0.4.2-SNAPSHOT</version>
+  <version>${revision}${changelist}</version>
   <packaging>hpi</packaging>
   <properties>
+    <revision>0.4.2</revision>
```
Comment (timja) [inline pom.xml]:
```suggestion
    <revision>0.5.0</revision>
```

File: pom.xml (hunk 5 above)
Comment (timja) [reply pom.xml:20]: It's required https://github.com/jenkinsci/theme-manager-plugin/blob/795dd3b2c8d2a7a89a7e86f47c231bd271fc4141/pom.xml#L17

Checkbox structure changed in 2.335 and there was an improvement in 2.336 required for theme manager.

Comment (timbrown5) [reply pom.xml:20]: Thanks, just checking if this version change will cause issues for #13 (just to know the order of merging).
The likelihood is that someone will need to update their PR. I will prefer yours you were first (unless #13 is stuck on an older version).

**Review by timbrown5 — APPROVED** [review]

**Pull request merged**